src/Eloquent/Model.php

```diff
@@ -199,6 +201,33 @@ public function getAttribute($key)
         return parent::getAttribute($key);
     }
 
+    /** @inheritdoc */
+    protected function transformModelValue($key, $value)
+    {
+        $value = parent::transformModelValue($key, $value);
+        if ($this->hasCast($key) && $value instanceof DateTimeInterface) {
+            $value->settings(array_merge($value->getSettings(), ['toStringFormat' => $this->getDateFormat()]));
```

Comment:
`settings` is a Carbon method. You can't assume any `DateTimeInterface` has this method.

Reply:
After I added [this](https://github.com/mongodb/laravel-mongodb/blob/698711ce63f8662e1579944ba47854fa0c5366e8/src/Eloquent/Model.php#L221-L223) to Model class, All date attributes in every type such as `\DateTime` or `string` and anything else are casting into a `Carbon` instance. If you have a date attribute but you don't cast it, it will be stored as an object and never reach this line.
However, should I check that `$value` is an instance of `Carbon` instead of `DateTimeInterface` to make sure the `settings` method is available?

Reply:
What should I say in the comment?

Reply:
> "date" cast convert values to Carbon.